src/flamenco/runtime/program/fd_system_program_nonce.c

```diff
@@ -919,6 +919,7 @@ fd_load_nonce_account( fd_exec_txn_ctx_t const *   txn_ctx,
   uchar const * instr_data = fd_txn_get_instr_data( txn_instr, txn_raw->raw );
   uchar const * instr_acct_idxs = fd_txn_get_instr_accts( txn_instr, txn_raw->raw );
 
+  /* TODO: Likely this check must be updated to support other nonce related instructions */
```

Comment:
what does this mean

Reply:
I assume he is referring to the change he did below where "has_nonce_account" is also checking to see if the instruction contained is one of the nonce instructions..

Reply:
fd_load_nonce_account is only used from fd_runtime_txn_lamports_per_signature.  This function looks to be the equiv of bank.rs:get_fee_for_message.  This calls check_message_for_nonce (bank.rs:3610) which calls get_durable_nonce in message/sanitized.rs which looks like it only works against a AdvanceNonceAccount instruction which matches the code.  So, I "believe" THIS code is correct and you should remove the todo

Now,  bank.rs:get_fee_for_message looks like it gives preference to the lamports_per_signature in the blockhash queue over the lamprots_per_signature contained in the nonce account.  It looks like fd_runtime_txn_lamports_per_signature does exactly the reverse...  This says we have a bug in fd_runtime_txn_lamports_per_signature